### PR TITLE
refactor: add plugin type definitions

### PR DESCRIPTION
**Pull request by guanbinrui**
## Description

Refactor the web3 tab to accept sub-tab registering from plugin definitions. Furthermore, the sub-tab could have resolved address names and the currently visiting identity as props. Finally, it brings any plugin that could have the ability to render a sub-tab on the Web3 tab.

## Under-going Tasks

+ [x] Move `Donations` and `Footprints` tab to the RSS3 plugin.
+ [x] Move `NFTs` tab to the collectible plugin.

image
image

closes #

### CODE REVIEW

File: packages/plugins/RSS3/src/base.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import type { Plugin } from '@masknet/plugin-infra'
+import { languages } from './locales'
+import { PLUGIN_NAME, PLUGIN_ID, PLUGIN_DESCRIPTION } from './constants'
+
+export const base: Plugin.Shared.Definition = {
+    ID: PLUGIN_ID,
+    icon: '🤔',
+    name: { fallback: PLUGIN_NAME },
```
Comment (Jack-Works) [inline packages/plugins/RSS3/src/base.ts:8]: Do you know those fields support i18n? By defining an "i18nKey".

image

File: packages/plugin-infra/src/types.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import type React from 'react'
 import type { Option, Result } from 'ts-results'
-import type { TypedMessage, TypedMessageTuple, ScopedStorage } from '@masknet/shared-base'
+import type { TypedMessage, TypedMessageTuple, ScopedStorage, ProfileIdentifier } from '@masknet/shared-base'
 import type { Emitter } from '@servie/events'
 import type { Web3Plugin } from './web3-types'
+import type { AddressName } from '../../shared/node_modules/@masknet/web3-shared-evm'
```
Comment (Jack-Works) [inline packages/plugin-infra/src/types.ts:6]: Wrong import path. I suggest you re-export `AddressName` from `@masknet/web3-shared-evm`

File: packages/plugin-infra/src/types.ts [hunk 3]
```diff
@@ -302,6 +307,48 @@ export namespace Plugin.SNSAdaptor {
...
+         */
+        label: I18NStringField | string
+        /**
+         * Used to order the sliders
+         */
+        priority: number
+        /**
+         * The injected UI
+         */
+        children: React.ReactNode
```
Comment (Jack-Works) [inline packages/plugin-infra/src/types.ts:333]: `InjectUI<{}>`

File: packages/plugins/RSS3/src/base.ts (hunk 1 above)
Comment (Jack-Works) [reply packages/plugins/RSS3/src/base.ts:8]: Now FileService plugin support i18n for those fields. I18n keys used in plugin definition must starts with `__`

File: packages/plugin-infra/src/types.ts [hunk 4]
```diff
@@ -199,6 +199,10 @@ export namespace Plugin.SNSAdaptor {
         CompositionDialogEntry?: CompositionDialogEntry
         /** This UI will be use when there is known badges. */
         CompositionDialogMetadataBadgeRender?: CompositionMetadataBadgeRender
+        /** This UI will be rendered as sliders on the profile page */
+        ProfileSliders?: ProfileSlider[]
```
Comment (Jack-Works) [inline packages/plugin-infra/src/types.ts:203]: Unused?

File: packages/plugins/RSS3/package.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    "@masknet/plugin-infra": "workspace:*",
+    "@masknet/shared": "workspace:*",
+    "@masknet/shared-base": "workspace:*",
+    "@masknet/theme": "workspace:*",
+    "@masknet/web3-providers": "workspace:*",
+    "@masknet/web3-shared-base": "workspace:*",
+    "@masknet/web3-shared-evm": "workspace:*",
+    "bignumber.js": "^9.0.1",
+    "date-fns": "^2.27.0",
+    "lodash-es": "^4.17.21",
```
Comment (Jack-Works) [inline packages/plugins/RSS3/package.json:17]: Do not install lodash-es. It's already installed at the top level.

File: packages/plugins/Debugger/package.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    "@masknet/plugin-infra": "workspace:*",
+    "@masknet/shared": "workspace:*",
+    "@masknet/shared-base": "workspace:*",
+    "@masknet/theme": "workspace:*",
+    "@masknet/web3-providers": "workspace:*",
+    "@masknet/web3-shared-base": "workspace:*",
+    "@masknet/web3-shared-evm": "workspace:*",
+    "bignumber.js": "^9.0.1",
+    "date-fns": "^2.27.0",
+    "lodash-es": "^4.17.21",
```
Comment (Jack-Works) [inline packages/plugins/Debugger/package.json:17]: Do not install this

File: packages/plugin-infra/src/types.ts (hunk 4 above)
Comment (guanbinrui) [reply packages/plugin-infra/src/types.ts:207]: Do it in the future.

**Pull request merged**